### ISSUE TITLE
Vertical mirroring mode does not work on Apollo Lake Linux with sample_vpp

### ISSUE THREAD

**Issue by ishmael1985** [Intel-Media-SDK/MediaSDK#90]
When specifying horizontal mirroring assuming NV12 to NV12 output as shown below, it seems to work perfectly:

```
ishmael@ishmael-desktop ~ $ /opt/intel/mediasdk/samples/sample_vpp -sw 640 -sh 360 -scc nv12 -dw 640 -dh 360 -dcc nv12 -iopattern d3d_to_d3d -mirror 1 -i /home/ishmael/Videos/nv12_640x360.yuv -o /home/ishmael/Videos/nv12_640x360_lr.yuv 
libva info: VA-API version 1.0.0
libva info: va_getDriverName() returns 0
libva info: User requested driver 'iHD'
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_0
libva info: va_openDriver() returns 0
Input format	NV12
Resolution	640x368
Crop X,Y,W,H	0,0,640,360
Frame rate	30.00
PicStruct	progressive
Output format	NV12
Resolution	640x368
Crop X,Y,W,H	0,0,640,360
Frame rate	30.00
PicStruct	progressive

Video Enhancement Algorithms
Deinterlace	OFF
Signal info	OFF
Scaling		OFF
Denoise		OFF
ProcAmp		OFF
DetailEnh	OFF
GamutMapping 	OFF
ColorSaturation	OFF
ContrastEnh  	OFF
SkinToneEnh  	OFF
MVC mode    	OFF
ImgStab    	OFF

IOpattern type               	d3d_to_d3d
Number of asynchronious tasks	1
Time stamps checking         	OFF
ROI checking                 	OFF

MediaSDK impl	hw
HW accelaration is enabled
MediaSDK ver	1.25
VPP started
Frame number: 809
VPP finished

Total frames 809 
Total time 18.61 sec 
Frames per second 43.462 fps 
```

However, when specifying vertical mirroring, I get the following segfault:

```
ishmael@ishmael-desktop ~ $ /opt/intel/mediasdk/samples/sample_vpp -sw 640 -sh 360 -scc nv12 -dw 640 -dh 360 -dcc nv12 -iopattern d3d_to_d3d -mirror 2 -i /home/ishmael/Videos/nv12_640x360.yuv -o /home/ishmael/Videos/nv12_640x360_tb.yuv 
libva info: VA-API version 1.0.0
libva info: va_getDriverName() returns 0
libva info: User requested driver 'iHD'
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_0
libva info: va_openDriver() returns 0

[ERROR], sts=MFX_ERR_UNSUPPORTED(-3), InitMemoryAllocator, pProcessor->pmfxVPP->Query failed at /home/ishmael/msdk/MediaSDK/samples/sample_vpp/src/sample_vpp_utils.cpp:614
Segmentation fault (core dumped)
```

Is this behaviour specific only to Apollo Lake platforms? How about Skylake, Broadwell, etc.?

**Comment by artem-shaporenko** [comment]
Vertical mirroring is not supported at a moment(you can see MFX_ERR_UNSUPPORTED), segfault - strange thing... Can you check were it is failing?

**Comment by ishmael1985** [comment]
Sure. Here goes:
```

ishmael@ishmael-desktop ~ $ gdb --args /opt/intel/mediasdk/samples/sample_vpp -sw 640 -sh 360 -scc nv12 -dw 640 -dh 360 -dcc nv12 -iopattern d3d_to_d3d -mirror 2 -i /home/ishmael/Videos/nv12_640x360.yuv -o /home/ishmael/Videos/nv12_640x360_tb.yuv 
GNU gdb (Ubuntu 7.11.1-0ubuntu1~16.5) 7.11.1
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /opt/intel/mediasdk/samples/sample_vpp...done.
(gdb) run
Starting program: /opt/intel/mediasdk/samples/sample_vpp -sw 640 -sh 360 -scc nv12 -dw 640 -dh 360 -dcc nv12 -iopattern d3d_to_d3d -mirror 2 -i /home/ishmael/Videos/nv12_640x360.yuv -o /home/ishmael/Videos/nv12_640x360_tb.yuv
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff64b0700 (LWP 10981)]
[New Thread 0x7ffff5caf700 (LWP 10982)]
[New Thread 0x7ffff54ae700 (LWP 10983)]
[New Thread 0x7ffff4cad700 (LWP 10984)]
libva info: VA-API version 1.0.0
libva info: va_getDriverName() returns 0
libva info: User requested driver 'iHD'
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_0
libva info: va_openDriver() returns 0

[ERROR], sts=MFX_ERR_UNSUPPORTED(-3), InitMemoryAllocator, pProcessor->pmfxVPP->Query failed at /home/ishmael/msdk/MediaSDK/samples/sample_vpp/src/sample_vpp_utils.cpp:614
[Thread 0x7ffff4cad700 (LWP 10984) exited]
[Thread 0x7ffff54ae700 (LWP 10983) exited]
[Thread 0x7ffff5caf700 (LWP 10982) exited]
[Thread 0x7ffff64b0700 (LWP 10981) exited]

Thread 1 "sample_vpp" received signal SIGSEGV, Segmentation fault.
0x00007ffff42aae80 in ?? ()
(gdb) bt
#0  0x00007ffff42aae80 in ?? ()
#1  0x00007ffff64b3f87 in vaDisplayIsValid (dpy=<optimized out>) at va.c:114
#2  0x00007ffff64b62d4 in vaDestroyContext (dpy=0x69d0a0, context=1073741824)
    at va.c:1199
#3  0x00007ffff670ae0c in MfxHwVideoProcessing::VAAPIVideoProcessing::Close (
    this=this@entry=0x6d36e0)
    at /home/ishmael/msdk/MediaSDK/_studio/shared/src/mfx_vpp_vaapi.cpp:191
#4  0x00007ffff670b4e7 in MfxHwVideoProcessing::VAAPIVideoProcessing::~VAAPIVideoProcessing (this=0x6d36e0, __in_chrg=<optimized out>)
    at /home/ishmael/msdk/MediaSDK/_studio/shared/src/mfx_vpp_vaapi.cpp:117
#5  0x00007ffff670b569 in MfxHwVideoProcessing::VAAPIVideoProcessing::~VAAPIVideoProcessing (this=0x6d36e0, __in_chrg=<optimized out>)
    at /home/ishmael/msdk/MediaSDK/_studio/shared/src/mfx_vpp_vaapi.cpp:119
#6  0x00007ffff671d1dc in std::default_delete<MfxHwVideoProcessing::DriverVideoProcessing>::operator() (this=<optimized out>, __ptr=<optimized out>)
    at /usr/include/c++/5/bits/unique_ptr.h:76
#7  std::unique_ptr<MfxHwVideoProcessing::DriverVideoProcessing, std::default_delete<MfxHwVideoProcessing::DriverVideoProcessing> >::reset (
    __p=<optimized out>, this=0x699290)
    at /usr/include/c++/5/bits/unique_ptr.h:344
#8  VPPHWResMng::Close (this=0x699290)
    at /home/ishmael/msdk/MediaSDK/_studio/shared/include/mfx_vpp_interface.h:406
---Type <return> to continue, or q <return> to quit---
#9  VPPHWResMng::~VPPHWResMng (this=0x699290, __in_chrg=<optimized out>)
    at /home/ishmael/msdk/MediaSDK/_studio/shared/include/mfx_vpp_interface.h:403
#10 VAAPIVideoCORE::~VAAPIVideoCORE (this=0x698eb0, __in_chrg=<optimized out>)
    at /home/ishmael/msdk/MediaSDK/_studio/shared/src/libmfx_core_vaapi.cpp:348
#11 0x00007ffff671d2d9 in VAAPIVideoCORE::~VAAPIVideoCORE (this=0x698eb0, 
    __in_chrg=<optimized out>)
    at /home/ishmael/msdk/MediaSDK/_studio/shared/src/libmfx_core_vaapi.cpp:358
#12 0x00007ffff66f9417 in mfx_core_ptr<VideoCORE>::reset (isNew=true, ptr=0x0, 
    this=0x698cd8)
    at /home/ishmael/msdk/MediaSDK/_studio/mfx_lib/shared/include/mfx_session.h:56
#13 _mfxSession::Cleanup (this=this@entry=0x698cd8)
    at /home/ishmael/msdk/MediaSDK/_studio/mfx_lib/shared/src/mfx_session.cpp:643
#14 0x00007ffff66f945e in _mfxSession::~_mfxSession (this=0x698cd8, 
    __in_chrg=<optimized out>)
    at /home/ishmael/msdk/MediaSDK/_studio/mfx_lib/shared/src/mfx_session.cpp:590
#15 0x00007ffff66f96a9 in _mfxSession_1_10::~_mfxSession_1_10 (this=0x698cd0, 
    __in_chrg=<optimized out>)
    at /home/ishmael/msdk/MediaSDK/_studio/mfx_lib/shared/src/mfx_session.cpp:795
---Type <return> to continue, or q <return> to quit---
#16 0x00007ffff66f0ae0 in MFXClose (session=0x698cd8)
    at /home/ishmael/msdk/MediaSDK/_studio/mfx_lib/shared/src/libmfxsw.cpp:280
#17 0x0000000000469d2c in MFX_DISP_HANDLE::UnLoadSelectedDLL() ()
#18 0x0000000000469d99 in MFX_DISP_HANDLE::Close() ()
#19 0x00000000004675d2 in MFXClose ()
#20 0x000000000042d0b5 in MFXVideoSession::Close (this=0x7ffffffeaf40)
    at /home/ishmael/msdk/MediaSDK/api/include/mfxvideo++.h:39
#21 WipeFrameProcessor (pProcessor=0x7ffffffeaf40)
    at /home/ishmael/msdk/MediaSDK/samples/sample_vpp/src/sample_vpp_utils.cpp:688
#22 0x000000000043071c in WipeResources (
    pResources=pResources@entry=0x7ffffffe5930)
    at /home/ishmael/msdk/MediaSDK/samples/sample_vpp/src/sample_vpp_utils.cpp:751
#23 0x00000000004369d8 in InitResources (
    pResources=pResources@entry=0x7ffffffe5930, 
    pParams=pParams@entry=0x7ffffffeaf70, 
    pInParams=pInParams@entry=0x7ffffffeb360)
    at /home/ishmael/msdk/MediaSDK/samples/sample_vpp/src/sample_vpp_utils.cpp:661
#24 0x0000000000418593 in main (argc=<optimized out>, argv=<optimized out>)
    at /home/ishmael/msdk/MediaSDK/samples/sample_vpp/src/sample_vpp.cpp:483
(gdb) 
```

**Comment by ishmael1985** [comment]
Sorry, I didn't compile Media SDK in debug mode

**Comment by artem-shaporenko** [comment]
Thanks, I believe that's enough info.

**Comment by adydychk** [comment]
@artem-shaporenko How can we classify this issue bug or feature request?

**Comment by ishmael1985** [comment]
Looks like this has been fixed by commit 46a73a3bfa317704b114328466140c54964303d2
